### PR TITLE
Avoid accumlation of supplied command line arguments 

### PR DESCRIPTION
**Pull request by aeslaughter**
@jiangwen84 This should fix the slow down

(refs #13342)

### CODE REVIEW

**Comment by jiangwen84** [comment]
@aeslaughter That is so great! Thanks for fixing this 👍

**Review by rwcarlsen — DISMISSED** [review]

**Comment by aeslaughter** [comment]
This is a WIP.

**Comment by aeslaughter** [comment]
@rwcarlsen I can't get anything to work, would you be willing to take a stab?

File: framework/include/parser/Parser.h [hunk 1]
```diff
@@ -146,6 +146,8 @@ class Parser : public ConsoleStreamInterface, public hit::Walker
 
   std::vector<std::string> listValidParams(std::string & section_name);
 
+  std::string hitCLIFilter(std::string appname, const std::vector<std::string> & argv);
```
Comment (rwcarlsen) [inline framework/include/parser/Parser.h]: doc comment explaining purpose of this function.

File: framework/include/parser/CommandLine.h [hunk 2]
```diff
@@ -114,7 +127,10 @@ class CommandLine
   std::map<std::string, Option> _cli_options;
 
 private:
+  /// CLI arguments that have been utilized
   std::set<int> _used_hiti;
+
+  /// CLI arguments supplied
```
Comment (rwcarlsen) [inline framework/include/parser/CommandLine.h]: "indices of CLI args that are HIT syntax parameters"

File: framework/include/parser/CommandLine.h [hunk 3]
```diff
@@ -114,7 +127,10 @@ class CommandLine
   std::map<std::string, Option> _cli_options;
 
 private:
+  /// CLI arguments that have been utilized
```
Comment (rwcarlsen) [inline framework/include/parser/CommandLine.h]: "indices of CLI args that have been marked as used"

**Review by rwcarlsen — CHANGES_REQUESTED** [review]
Looks okay.  We really should try to consolidate the redundant logic in Parser.C and CommandLine.C for characterizing command line args.  Did you look into trying that?

**Comment by aeslaughter** [comment]
@rwcarlsen I did think about removing the similar code, but the differences are significant enough that it wasn't trivial and I (and you) have already spent way too much time on this, but I created issue #13518.

**Comment by aeslaughter** [comment]
@rwcarlsen I did think about removing the similar code, but the differences are significant enough that it wasn't trivial and I (and you) have already spent way too much time on this, but I created issue #13518.

**Review by rwcarlsen — APPROVED** [review]

**Pull request merged**